src/plugins/hetero/src/compiled_model.cpp

```diff
@@ -18,45 +18,29 @@
 #include "openvino/util/common_util.hpp"
 #include "plugin.hpp"
 #include "properties.hpp"
+#include "subgraph_collector.hpp"
 #include "xml_parse_utils.h"
 
-template <typename T>
-using NodeMap = std::unordered_map<std::shared_ptr<ov::Node>, T>;
-
-namespace {
-
-template <typename Set>
-static Set intersection(const Set& lhs, const Set& rhs) {
-    Set result;
-    const auto& minSizeSet = (lhs.size() < rhs.size()) ? lhs : rhs;
-    const auto& maxSizeSet = (lhs.size() >= rhs.size()) ? lhs : rhs;
-    for (auto&& val : minSizeSet)
-        if (maxSizeSet.find(val) != maxSizeSet.end())
-            result.insert(val);
-    return result;
-}
-
-template <typename Set>
-static bool intersects(const Set& lhs, const Set& rhs) {
-    const auto& minSizeSet = (lhs.size() < rhs.size()) ? lhs : rhs;
-    const auto& maxSizeSet = (lhs.size() >= rhs.size()) ? lhs : rhs;
-    for (auto&& val : minSizeSet)
-        if (maxSizeSet.find(val) != maxSizeSet.end())
-            return true;
-    return false;
-}
-}  // namespace
-
 ov::hetero::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
                                          const std::shared_ptr<const ov::IPlugin>& plugin,
                                          const Configuration& cfg)
     : ov::ICompiledModel(model, plugin),
       m_cfg(cfg),
       m_name(model->get_friendly_name()),
       m_loaded_from_cache(false) {
-    bool dumpDotFile = m_cfg.dump_graph;
+    try {
+        compile_model(model);
+    } catch (const std::exception& e) {
+        OPENVINO_THROW("Standard exception from compilation library: ", e.what());
+    } catch (...) {
+        OPENVINO_THROW("Generic exception is thrown");
+    }
```

Comment:
Such wrappers were intentionally removed because all exceptions will be converted to `ov::Exception` in `Core.compile_model()`